### ISSUE TITLE
[v4] writerMarks don't work in Block with preview: text

### ISSUE THREAD

**Issue by rasteiner** [k-next/kirby#5477]
## Description
When custom marks are used on a block with `preview: text`, clicking on the custom mark's button doesn't do anything. 
But when opening the same block in the "Edit drawer", the custom mark button works.

**Expected behavior**  
Custom marks should work in block which use `preview: text`

**Screenshots**  
![custommarks](https://github.com/getkirby/kirby/assets/6684137/0694a678-c8a8-4432-94f7-af4ac66a7fda)

## To reproduce

1. Recover the minimal setup from https://github.com/getkirby/kirby/issues/5456#issuecomment-1666442697
2. Change the blueprint to:
    ```yml
    layout:
      type: layout
      fieldsets:
        a:
          # wysiwyg: true # comment this
          preview: text # change this
          fields:
            text:
              type: writer
              nodes: false
              marks:
                - highlight
    ```
3. Try using the custom writer mark on the preview
4. Open the edit drawer and use it there

## Your setup
**Kirby Version**  
Kirby 4 alpha 6

**Console output**  
Nothing :(
The function I define in `commands()` is run, but `this.toggle()` doesn't seem to do anything visible. 

**Your system (please complete the following information)**
 - Device: Windows PC
 - OS: Windows 11
 - Browser: Chrome
 - Version: 115

**Comment by rasteiner** [comment]
PS:

actually, it's not about `preview: text`. I've tried with `preview: fields` and it still doesn't work :/

**Comment by rasteiner** [comment]
Don't know if related... If I open the drawer and then exit it (even without changing anything) and then try my button again on the preview, the field crashes with this message in the console:

## on chrome
```console
index.js:1 TypeError: Cannot read properties of null (reading 'matchesNode')
    at Co.updateStateInner (vendor.js:1:178396)
    at Co.updateState (vendor.js:1:177708)
    at _i.dispatchTransaction (index.js:1:210764)
    at Co.dispatch (vendor.js:1:183157)
    at vendor.js:1:193163
    at _i.toggleMark (index.js:1:214526)
    at Ci.toggle (index.js:1:214862)
    at index.js?1691761820:254:27
    at l.<computed> (index.js:1:205256)
    at _i.command (index.js:1:208360)
```

<img width="360" alt="image" src="https://github.com/getkirby/kirby/assets/6684137/8278e9e9-b758-4a8c-accf-2606057e90d4">

## on firefox 
the field crashes with the message `this.docView is null`:
```console
TypeError: this.docView is null
    updateStateInner http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/vendor.js:1
    updateState http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/vendor.js:1
    dispatchTransaction http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/index.js:1
    dispatch http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/vendor.js:1
    ls http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/vendor.js:1
    toggleMark http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/index.js:1
    toggle http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/index.js:1
    commands http://localhost/media/plugins/index.js?1691761820:254
    t http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/index.js:1
    command http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/index.js:1
    onCommand http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/index.js:1
    VueJS 4
    command http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/index.js:1
    mousedown http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/index.js:1
    VueJS 26
    updateStateInner http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/vendor.js:1
    updateState http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/vendor.js:1
    dispatchTransaction http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/index.js:1
    dispatch http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/vendor.js:1
    ko http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/vendor.js:1
    domObserver http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/vendor.js:1
    flush http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/vendor.js:1
    onSelectionChange http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/vendor.js:1
    connectSelection http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/vendor.js:1
    qr http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/vendor.js:1
    updateStateInner http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/vendor.js:1
    updateState http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/vendor.js:1
    dispatchTransaction http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/index.js:1
    dispatch http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/vendor.js:1
    ko http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/vendor.js:1
    domObserver http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/vendor.js:1
    flush http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/vendor.js:1
    onSelectionChange http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/vendor.js:1
    connectSelection http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/vendor.js:1
    qr http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/vendor.js:1
    focus http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/vendor.js:1
    focus http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/index.js:1
    setTimeout handler*focus http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/index.js:1
    focus http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/index.js:1
    focus http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/index.js:1
    focus http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/index.js:1
    focus http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/index.js:1
    onClose http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/index.js:1
    emit http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/index.js:1
    close http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/index.js:1
    close http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/index.js:1
    close http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/index.js:1
    VueJS 4
    onClose http://localhost/media/panel/9917c401e367057bbf3c38bffc644991/js/index.js:1
    VueJS 3
```

**Comment by rasteiner** [comment]
meh... cannot replicate on "not imported content". I guess I'm doing something wrong when importing the blocks... Will open new issue if there is anything..

**Comment by rasteiner** [comment]
whatever it is, it's actually fixed by the dist folder of #5456

**Comment by distantnative** [comment]
@rasteiner Yes I think there can be very weird bugs currently with custom marks and nodes because my first implementation introduced some nasty reactivity issues where different writer instances would be mixed up.

But glad the dist files there fixed yours too. Means that my fix might be a good one.

**Comment by rasteiner** [comment]
yes, I think it turned out to be exactly the same issue as #5456, I've just misinterpreted its title, I though "latest" referred to "the last editor to be used", but it actually meant "the last editor to be instanced". 

That's why it worked in the drawer (that's always the last one to be instanced), and it also explains why it crashed when the drawer was closed. That's also why it didn't seem to work on "imported content" (which was way longer, with many blocks and instances, the last one being offscreen).

Silly me 😅
